renderdoc/driver/vulkan/wrappers/vk_resource_funcs.cpp

```diff
@@ -3121,6 +3121,123 @@ void WrappedVulkan::vkSetDeviceMemoryPriorityEXT(VkDevice device, VkDeviceMemory
+      {
+        live = GetResourceManager()->GetNonDispWrapper(acc)->id;
+
+        // destroy this instance of the duplicate, as we must have matching create/destroy
+        // calls and there won't be a wrapped resource hanging around to destroy this one.
+        ObjDisp(device)->DestroyAccelerationStructureKHR(Unwrap(device), acc, NULL);
+
+        // whenever the new ID is requested, return the old ID, via replacements.
+        GetResourceManager()->ReplaceResource(AccelerationStructure,
+                                              GetResourceManager()->GetOriginalID(live));
+      }
+      else
+      {
+        live = GetResourceManager()->WrapResource(Unwrap(device), acc);
+        GetResourceManager()->AddLiveResource(AccelerationStructure, acc);
+
+        m_CreationInfo.m_AccelerationStructure[live].Init(GetResourceManager(), m_CreationInfo,
+                                                          &CreateInfo);
+      }
+    }
+
+    AddResource(AccelerationStructure, ResourceType::Buffer, "AccelerationStructure");
+    DerivedResource(device, AccelerationStructure);
+    DerivedResource(CreateInfo.buffer, AccelerationStructure);
+  }
+
+  return true;
+}
+
+VkResult WrappedVulkan::vkCreateAccelerationStructureKHR(
+    VkDevice device, const VkAccelerationStructureCreateInfoKHR *pCreateInfo,
+    const VkAllocationCallbacks *pAllocator, VkAccelerationStructureKHR *pAccelerationStructure)
+{
+  VkAccelerationStructureCreateInfoKHR unwrappedInfo = *pCreateInfo;
+  unwrappedInfo.buffer = Unwrap(unwrappedInfo.buffer);
+  VkResult ret;
+  SERIALISE_TIME_CALL(ret = ObjDisp(device)->CreateAccelerationStructureKHR(
+                          Unwrap(device), &unwrappedInfo, pAllocator, pAccelerationStructure));
+
+  if(ret == VK_SUCCESS)
+  {
+    ResourceId id = GetResourceManager()->WrapResource(Unwrap(device), *pAccelerationStructure);
+
+    if(IsCaptureMode(m_State))
+    {
+      Chunk *chunk = NULL;
+
+      {
+        CACHE_THREAD_SERIALISER();
+
+        SCOPED_SERIALISE_CHUNK(VulkanChunk::vkCreateAccelerationStructureKHR);
+        Serialise_vkCreateAccelerationStructureKHR(ser, device, pCreateInfo, NULL,
+                                                   pAccelerationStructure);
+
+        chunk = scope.Get();
+      }
+
+      VkResourceRecord *bufferRecord = GetRecord(pCreateInfo->buffer);
+
+      VkResourceRecord *record = GetResourceManager()->AddResourceRecord(*pAccelerationStructure);
+      record->AddChunk(chunk);
+      record->AddParent(bufferRecord);
+
+      // store the base resource
+      record->baseResource = bufferRecord->GetResourceID();
+      record->baseResourceMem = bufferRecord->baseResource;
+      record->dedicated = bufferRecord->dedicated;
+      record->resInfo = bufferRecord->resInfo;
+      record->storable = bufferRecord->storable;
+      record->memOffset = bufferRecord->memOffset + pCreateInfo->offset;
+      record->memSize = bufferRecord->memSize - pCreateInfo->offset;
```

Comment:
I think this `memSize` should be the size of the AS - `pCreateInfo->size`, not the whole rest of the buffer.

This will be relevant for the memory reference tracking later.

Reply:
Just to clarify, are you saying that the size of the AS is `pCreateInfo->size` or that I should use `(size of AS) - pCreateInfo->size`?  If the latter, where is the AS size defined?